### PR TITLE
Feature/cache improvements 2

### PR DESCRIPTION
**Pull request by rPraml**
Splitted in a separete PR - this add additional features
- cache findIds
- return unmodifiable collections or CopyOnFirstWrite-lists.

### CODE REVIEW

**Comment by rbygrave** [comment]
Hi, So yes this all generally makes sense.  I'm not sure we should deprecate setUseQueryCache(boolean) ... just yet.  We may not ever end up using CacheMode with the bean cache given how  that works.  That means the CacheMode really only adds the RECACHE option.

I'll ponder a bit more tomorrow.

File: src/main/java/io/ebean/common/BeanList.java [hunk 1]
```diff
@@ -541,4 +541,11 @@ public int previousIndex() {
     }
 
   }
+
+  @Override
+  public BeanCollection<E> getShallowCopy() {
+    BeanList<E> copy = new BeanList<>(new CopyOnFirstWriteList<>(list));
+    copy.setFromOriginal(copy);
```
Comment (rbygrave) [inline src/main/java/io/ebean/common/BeanList.java]: copy.setFromOriginal(this);  -- should use this here ?

**Comment by rPraml** [comment]
... still in progress. I will resolve merge conflicts and so on

**Comment by rbygrave** [comment]
So this is ready to merge in right?

**Comment by rPraml** [comment]
@rbygrave this is ready to merge in.

**Pull request merged**